src/topology/instances/ennreal.lean

```diff
@@ -696,24 +702,18 @@ show summable (λb, g' b : β → ℝ), from nnreal.summable_coe.2 this
 the sequence of partial sum converges to `r`. -/
 lemma has_sum_iff_tendsto_nat_of_nonneg {f : ℕ → ℝ} (hf : ∀i, 0 ≤ f i) (r : ℝ) :
   has_sum f r ↔ tendsto (λn:ℕ, ∑ i in finset.range n, f i) at_top (𝓝 r) :=
-⟨has_sum.tendsto_sum_nat,
-  assume hfr,
-  have 0 ≤ r := ge_of_tendsto hfr $ univ_mem_sets' $ assume i,
-    show 0 ≤ ∑ j in finset.range i, f j, from finset.sum_nonneg $ assume i _, hf i,
-  let f' (n : ℕ) : nnreal := ⟨f n, hf n⟩, r' : nnreal := ⟨r, this⟩ in
-  have f_eq : f = (λi:ℕ, (f' i : ℝ)) := rfl,
-  have r_eq : r = r' := rfl,
-  begin
-    rw [f_eq, r_eq, nnreal.has_sum_coe, nnreal.has_sum_iff_tendsto_nat, ← nnreal.tendsto_coe],
-    simp only [nnreal.coe_sum],
-    exact hfr
-  end⟩
+begin
+  lift f to ℕ → ℝ≥0 using hf,
```

Comment:
I think this trick is worth mentioning in some module docstring, probably in the file defining `ℝ≥0`.